lib/encryption/key_manager.dart

```diff
@@ -303,7 +303,8 @@ class KeyManager {
         continue;
       }
       final userDeviceKeyIds = deviceKeyIds[device.userId] ??= <String, bool>{};
-      userDeviceKeyIds[deviceId] = !device.encryptToDevice;
+      userDeviceKeyIds[deviceId] =
+          !device.encryptToDevice && device.userId == client.userID!;
```

Comment:
I'm not sure what this method actually does. It has no description and a non intuitive name. But I guess it is to convert a list of devices keys to a map to know if we want to share keys with the specific device of a user. However we don't know for waht purpose we use this. If this is for normal key sharing we need to send the keys to other users as well